src/test/e2e/oidc.e2e.ts

```diff
@@ -0,0 +1,150 @@
+  let oidcMockProviderEndpointAccesses: Record<string, number>;
+
+  let i = 0;
+  let tmpdir: string;
+  let cluster: MongoCluster;
+  let connectionString: string;
+
+  before(async function () {
+    if (process.platform !== 'linux') {
+      // return;
+      this.skip();
+    }
+
+    oidcMockProviderEndpointAccesses = {};
+    oidcMockProviderConfig = {
+      getTokenPayload(metadata: Parameters<typeof getTokenPayload>[0]) {
+        return getTokenPayload(metadata);
+      },
+      overrideRequestHandler(url, req, res) {
+        const { pathname } = new URL(url);
+        oidcMockProviderEndpointAccesses[pathname] ??= 0;
+        oidcMockProviderEndpointAccesses[pathname]++;
+        return overrideRequestHandler?.(url, req, res);
+      },
+    };
+    oidcMockProvider = await OIDCMockProvider.create(oidcMockProviderConfig);
+
+    tmpdir = path.join(
+      os.tmpdir(),
+      `vscode-oidc-${Date.now().toString(32)}-${++i}`
+    );
+    await fs.mkdir(path.join(tmpdir, 'db'), { recursive: true });
+    const serverOidcConfig = {
+      issuer: oidcMockProvider.issuer,
+      clientId: 'testServer',
+      requestScopes: ['mongodbGroups'],
+      authorizationClaim: 'groups',
+      audience: 'resource-server-audience-value',
+      authNamePrefix: 'dev',
+    };
+
+    cluster = await startTestServer({
+      version: '7.0.x',
+      downloadOptions: { enterprise: true },
+      args: [
+        '--setParameter',
+        'authenticationMechanisms=SCRAM-SHA-256,MONGODB-OIDC',
+        // enableTestCommands allows using http:// issuers such as http://localhost
+        '--setParameter',
+        'enableTestCommands=true',
+        '--setParameter',
+        `oidcIdentityProviders=${JSON.stringify([serverOidcConfig])}`,
+      ],
+    });
+
+    const cs = new ConnectionString(cluster.connectionString);
+    cs.searchParams.set('authMechanism', 'MONGODB-OIDC');
+
+    connectionString = cs.toString();
+  });
+
+  after(async function () {
+    await cluster?.close();
+    await oidcMockProvider?.close();
+    if (tmpdir) await fs.rmdir(tmpdir, { recursive: true });
+  });
+
+  test('should connect successfully with a connection string', async function () {
+    const workbench = await browser.getWorkbench();
+
+    // Connect with OIDC connection string
+    const connectionStringInput = await workbench.executeCommand(
+      'MongoDB: Connect with Connection String...'
+    );
+    await connectionStringInput.wait();
+    await connectionStringInput.setText(connectionString);
+    await connectionStringInput.confirm();
+    await browser.waitUntil(invisibilityOf(connectionStringInput.elem));
+
+    // Check if we have connection successful notification
+    await browser.waitUntil(
+      async () => {
+        const notifs = await workbench.getNotifications();
+        const messages = await Promise.all(notifs.map((n) => n.getMessage()));
+        return messages.includes('MongoDB connection successful.');
```

Comment:
Is there some easy way of verifying that we connected as an authenticated user (and in particular that the connection attempt did not accidentally succeed because it is fully unauthenticated and makes use of the MongoDB localhost exception)?

In Compass and mongosh we use `connectionStatus` and/or check that the OIDC IDP has actually issued a token by looking at the contents of `oidcMockProviderEndpointAccesses`, those might be good things to try here

Reply:
yea in addition to what is suggested I also wanted to include the authInfo check which unfortunately due to limitations being posed right now, I am unable to. Will be looking into this today but thank you for this ⭐